test/library/stark/test_ramsey_xy.py

```diff
@@ -0,0 +1,125 @@
+from qiskit.providers.fake_provider import FakeHanoiV2
+
+from qiskit_experiments.library import StarkRamseyXY
+
+
+class TestStarkRamseyXY(QiskitExperimentsTestCase):
+    """Test case for the Stark Ramsey XY experiment.
+
+    Because the analysis is identical to the RamseyXY, integration test and
+    test for analysis class is skipped here. These must be covered by
+    test/library/calibration/test_ramsey_xy.py
+    """
+
+    def test_calibration_with_positive_amp(self):
+        """Test Stark frequency shift chooses the proper sign with positive amplitude."""
+        backend = FakeHanoiV2()
+        exp = StarkRamseyXY(
+            qubit=0,
+            stark_amp=0.1,  # positive amplitude
+            backend=backend,
+            stark_sigma=15e-9,
+            stark_risefall=2,
+            stark_freq_offset=80e6,
+        )
+        param_ram_x, _ = exp.parameterized_circuits()
+        freq = backend.qubit_properties(0).frequency - 80e6  # negative frequency shift
+        granularity = backend.target.granularity
+        dt = backend.dt
+        duration = granularity * int(round(4 * 15e-9 / dt / granularity))
+        sigma = duration / 4
+
+        with pulse.build() as ref_schedule:
+            pulse.set_frequency(freq, pulse.DriveChannel(0))
+            pulse.play(
+                pulse.Gaussian(duration=duration, amp=0.1, sigma=sigma), pulse.DriveChannel(0)
+            )
+
+        test_schedule = param_ram_x.calibrations["StarkV"][(0,), ()]
+        self.assertEqual(test_schedule, ref_schedule)
+
+    def test_calibration_with_negative_amp(self):
+        """Test Stark frequency shift chooses the proper sign with negative amplitude."""
+        backend = FakeHanoiV2()
+        exp = StarkRamseyXY(
+            qubit=0,
+            stark_amp=-0.1,  # negative amplitude
+            backend=backend,
+            stark_sigma=15e-9,
+            stark_risefall=2,
+            stark_freq_offset=80e6,
+        )
+        param_ram_x, _ = exp.parameterized_circuits()
+        freq = backend.qubit_properties(0).frequency + 80e6  # positive frequency shift
+        granularity = backend.target.granularity
+        dt = backend.dt
+        duration = granularity * int(round(4 * 15e-9 / dt / granularity))
+        sigma = duration / 4
+
+        with pulse.build() as ref_schedule:
+            pulse.set_frequency(freq, pulse.DriveChannel(0))
+            pulse.play(
+                pulse.Gaussian(duration=duration, amp=0.1, sigma=sigma), pulse.DriveChannel(0)
+            )
+
+        test_schedule = param_ram_x.calibrations["StarkV"][(0,), ()]
+        self.assertEqual(test_schedule, ref_schedule)
+
+    def test_gen_delays(self):
+        """Test generating delays with experiment options."""
+        exp = StarkRamseyXY(
+            qubit=0,
+            stark_amp=0.1,
+            min_freq=1e6,
+            max_freq=50e6,
+        )
+        test_delays = exp.delays()
+        ref_delays = np.arange(0, 1e-6, 1e-8)
```

Comment:
This might help someone looking at this later:

```suggestion
        min_freq = 1e6
        max_freq = 50e6
        exp = StarkRamseyXY(
            qubit=0,
            stark_amp=0.1,
            min_freq=min_freq,
            max_freq=max_freq,
        )
        test_delays = exp.delays()
        ref_delays = np.arange(0, 1 / min_freq, 1 / max_freq / 2)
```